### PR TITLE
feat: introduce CoreHealthSystem to separate API from default features

### PR DESCRIPTION
**Pull request by skaldarnar**
This is just another small step towards restructuring this whole module. Expect things to move around again in later PRs.

Depends on https://github.com/Terasology/Health/pull/80 and https://github.com/Terasology/Health/pull/81.

### CODE REVIEW

File: src/main/java/org/terasology/module/health/core/CoreHealthSystem.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.module.health.core;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/module/health/core/CoreHealthSystem.java:4]: I've put this in a separate package, although I'm not sure about it. My goal is to separate the API of Health other modules can and should integrate with from the _default logic_ served by this module. 

Other aspects that I see in this system are the short timeout before regeneration when an entity is hit. These default feature themselves use the API and serve as example implementations (at least that's my idea - "eat your own dog food" :dog: )

File: src/main/java/org/terasology/module/health/systems/RegenAuthoritySystem.java [hunk 2]
```diff
@@ -111,6 +98,8 @@ private void invokeRegenOperations(long currentWorldTime) {
                 .forEach(regenEntity -> {
                     RegenComponent regen = regenEntity.getComponent(RegenComponent.class);
                     regenSortedByTime.remove(regen.soonestEndTime, regenEntity);
+                    //TODO: Add a flag to indicate the regeneration id should be removed as soon as the entity is restored to full health?
+                    //      On the other hand, a system can react to the OnFullyHealedEvent and unregister the regeneration id if desired.
```
Comment (skaldarnar) [inline src/main/java/org/terasology/module/health/systems/RegenAuthoritySystem.java:102]: I tend toward removing this piece of logic to remove "completed" regeneration actions, but I'm not sure what the _usual use case_ would be here. I can imagine having a flag to indicate that the regen action is a "only-once" action and should be removed, but I'd like to explore how complicated the solution via `OnFullyHealedEvent` turns out.

File: src/main/java/org/terasology/module/health/systems/RegenAuthoritySystem.java [hunk 3]
```diff
@@ -83,26 +83,13 @@ public void update(float delta) {
         }
     }
 
-
     /**
      * @param currentWorldTime the current in-game time in ms
      */
     private void invokeRegenOperations(long currentWorldTime) {
         // Contains all the entities with current time crossing EndTime
-        List<EntityRef> entitiesWithExpiringRegenActions = new LinkedList<>();
```
Comment (skaldarnar) [inline src/main/java/org/terasology/module/health/systems/RegenAuthoritySystem.java:92]: This was plainly refactored using IntelliJ's "extract this" action.

**Comment by skaldarnar** [comment]
[Module Overhaul - Health](https://trello.com/c/zMUpksot/92-module-overhaul-health)

**Review by jdrueckert — APPROVED** [review]

**Pull request merged**